comp/core/workloadmeta/store.go

```diff
@@ -450,6 +450,41 @@ func (w *workloadmeta) Reset(newEntities []Entity, source Source) {
 	w.Notify(events)
 }
 
+func (w *workloadmeta) validatePushEvents(events []PushEvent) error {
```

Comment:
Should we also validate the entity `Kind` ?

Reply:
I think so, otherwise the caller of this function will not realize it made a mistake because notify adds the entities to the store asynchronously.

Reply:
Please check Possible Drawbacks / Trade-offs in the PR description.